Versi [Google Collab](https://colab.research.google.com/drive/10-1vxoQ5Mxp654Hs0b88AJ2ub92p0Z-B#scrollTo=qLb2bWu8mkrF)

In [ ]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [ ]:
#Create DataFrame

data = {'TransactionAmount': [100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]}

df = pd.DataFrame(data)
print(df)

    TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65


In [ ]:
#Central Tendency

mean = df['TransactionAmount'].mean()
median = df['TransactionAmount'].median()
mode = df['TransactionAmount'].mode()[0]

print("Mean: ", mean)
print("Median: ", median)
print("Mode: ", mode)

Mean:  102.6
Median:  100.0
Mode:  100


Berdasarkan hasil output di atas, dapat disimpulkan bahwa nilai mean, median, dan mode dari data ‘TransactionAmount’ adalah 102.6, 100.0, dan 100, secara berturut-turut. Hal ini menunjukkan bahwa data memiliki kecenderungan yang sama dan terdistribusi secara merata.

In [ ]:
#Variability

var = df['TransactionAmount'].var()
std = df['TransactionAmount'].std()
max_val = df['TransactionAmount'].max()
min_val = df['TransactionAmount'].min()
q1 = df['TransactionAmount'].quantile(0.25)
median = df['TransactionAmount'].quantile(0.5)
q3 = df['TransactionAmount'].quantile(0.75)

print("Variance: ", var)
print("Standard Deviation: ", std)
print("Max Value: ", max_val)
print("Min Value: ", min_val)
print("Q1: ", q1)
print("Median: ", median)
print("Q3: ", q3)

Variance:  1012.7499999999999
Standard Deviation:  31.823733281939123
Max Value:  150
Min Value:  50
Q1:  85.0
Median:  100.0
Q3:  120.0


Berdasarkan hasil output di atas, dapat disimpulkan bahwa nilai variansi, standar deviasi, max, min, serta kuartil dari data ‘TransactionAmount’ adalah 1012.749, 31.823, 150, 50, 85, 100, dan 120, secara berturut-turut. Hal ini dapat digunakan untuk memahami sebaran data ‘TransactionAmount’ dan melihat outlier (jika ada).

In [ ]:
from scipy.stats import ttest_1samp

# menghitung t-statistics dan p-value
t_statistic, p_value = ttest_1samp(df['TransactionAmount'], 100) # define h0 = 100, h1 = < 100 >

print("t-statistic: ", t_statistic)
print("p-value: ", p_value)

# melakukan one sample t-test
alpha = 0.05 # define alpha 0.05
if p_value < alpha:
    print("Reject null hypothesis (H0)")
else:
    print("Fail to reject null hypothesis (H0)")

t-statistic:  0.4085001556802841
p-value:  0.6865284813438117
Fail to reject null hypothesis (H0)


**Kesimpulan**

Berdasarkan hasil output di atas, dapat disimpulkan bahwa nilai t-statistic adalah 0.408, dan p-value adalah 0.686. Dalam hal ini, karena p-value > alpha (0.05), maka kita gagal menolak hipotesis nol (null hypothesis) yang menyatakan bahwa penjualan rata-rata (nilai H0) adalah USD 100 per transaksi. Dengan kata lain, hasil uji statistik menunjukkan bahwa tidak ada perbedaan yang signifikan antara penjualan rata-rata saat ini dengan nilai H0 sebelumnya.

In [ ]:
#Critical region result

from scipy.stats import t

n = len(df)
alpha = 0.05
df_degrees = n - 1

# Menghitung t-score dari tabel distribusi t
t_score = t.ppf(alpha, df_degrees)

# Menghitung critical region (nilai H1)
h1 = (df['TransactionAmount'].mean() < 100 - t_score * df['TransactionAmount'].std()) or (df['TransactionAmount'].mean() > 100 + t_score * df['TransactionAmount'].std())

print("Critical Region (H1): ", h1)

Critical Region (H1):  True


### Catatan :
stats.t.ppf(1-0.05, 24)
coba keluarkan nilai p-value, dan simpulkan apakah diatas nilai alpha atau tidak, kemudian simpulkan apakah tolak H0 atau gagal menolak H0 ???

Notes : untuk penerapan p-value, alpha dan simpulan h0, sudah dikalkulasikan pada notes sebelum ini.


Nilai H1 adalah True, artinya terdapat perbedaan yang signifikan antara penjualan rata-rata saat ini dengan nilai H0 sebelumnya. Hal ini dapat diartikan bahwa pelatihan yang telah diberikan kepada pekerja sales telah berhasil dalam meningkatkan penjualan per transaksi, sehingga rata-rata penjualan saat ini lebih tinggi dari nilai H0 sebelumnya ($100 per transaksi). Dalam hal ini, perusahaan X dapat melanjutkan upaya untuk meningkatkan penjualan dengan cara melatih pekerja sales lebih lanjut atau memberikan insentif yang lebih besar bagi pekerja sales yang berhasil meningkatkan penjualannya.